tests/org.eclipse.ui.editors.tests/src/org/eclipse/ui/editors/tests/FindNextActionTest.java

```diff
@@ -0,0 +1,253 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.ui.editors.tests;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.fail;
+
+import java.util.ResourceBundle;
+
+import org.junit.After;
+import org.junit.Test;
+import org.osgi.framework.FrameworkUtil;
+
+import org.eclipse.core.runtime.CoreException;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+
+import org.eclipse.core.filebuffers.tests.ResourceHelper;
+
+import org.eclipse.jface.dialogs.IDialogSettings;
+import org.eclipse.jface.viewers.ISelectionProvider;
+
+import org.eclipse.jface.text.Document;
+import org.eclipse.jface.text.TextSelection;
+
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.IWorkbenchPartSite;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.ide.IDE;
+import org.eclipse.ui.internal.findandreplace.HistoryStore;
+
+import org.eclipse.ui.texteditor.AbstractTextEditor;
+import org.eclipse.ui.texteditor.FindNextAction;
+
+
+public class FindNextActionTest {
+	private IFile fFile;
+
+	private IEditorPart fEditor;
+
+	private int fCount;
+
+	private FindNextAction action;
+
+	private static final String BUNDLE_FOR_CONSTRUCTED_KEYS= "org.eclipse.ui.texteditor.ConstructedEditorMessages";//$NON-NLS-1$
+
+	private static ResourceBundle fgBundleForConstructedKeys= ResourceBundle.getBundle(BUNDLE_FOR_CONSTRUCTED_KEYS);
+
+	public void openEditorAndFindNextAction(String content, boolean direction) {
+		try {
+			IFolder folder= ResourceHelper.createFolder("EncodingChangeTestProject/EncodingChangeTests/");
+			fFile= ResourceHelper.createFile(folder, "file" + fCount + ".txt", content);
+			fFile.setCharset(null, null);
+			fCount++;
+		} catch (CoreException e) {
+			fail();
```

Comment:
Please no such empty fail() calls, it becomes a nightmare figuring out what happened when such a fail occurs. As a bare minimum pass e.getMessage() but it's even better to not catch the exception but let the method throws it and let JUnit fail the test itself with the message and full stack.

Reply:
Yes, these fail statements are actually completely unnecessary. I am currently updating the PR accordingly (and improving it in general).